var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +62,21 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling
+    depends_on(
+        "hiop+deepchecking build_type=RelWithDebInfo", when="+hiop build_type=RelWithDebInfo"
+    )
+    depends_on("hiop~deepchecking  build_type=Release ", when="+hiop build_type=Release ")
+
+    # Control the package's build-type depending on the release or debug flag
+    for pkg in ["raja", "umpire", "magma", "camp", "ginkgo"]:
+        depends_on("{0} build_type=Release".format(pkg), when="build_type=Release")
+        depends_on("{0} build_type=RelWithDebInfo".format(pkg), when="build_type=RelWithDebInfo")
```

Comment:
```suggestion
        depends_on("{0} build_type=Release".format(pkg), when="+{0} build_type=Release".format(pkg))
        depends_on("{0} build_type=RelWithDebInfo".format(pkg), when="+{0} build_type=RelWithDebInfo".format(pkg))
```

I'm not as adept at reading audit logs as @alalazo, but I think this is the correct logic, and a upon re-reading I think I found the bug

Reply:
Turns out there isn't actually a variant that maps directly to each package... @ryandanehy can you create a tuple list that pairs each package to the variant combination that controls it, and then modify this loop accordingly?

Reply:
I believe it just ginkgo that doesn't have a relwithDebug build type. @CameronRutherford so I think I can just do a depends on statements for ginkgo separately rather than have a tuple list.

Reply:
All packages have the same set of build types, so I am not sure why ginkgo is different - perhaps we need to hard code for that package. Style is failing, but additionally we see errors in the audit:

```
1. exago: wrong variant in "depends_on" directive
    the variant 'umpire' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/exago/package.py
2. exago: wrong variant in "depends_on" directive
    the variant 'magma' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/exago/package.py
3. exago: wrong variant in "depends_on" directive
    the variant 'camp' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/exago/package.py
4. exago: wrong variant in "depends_on" directive
    the variant 'ginkgo' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/exago/package.py
5. exago: wrong variant used for a dependency in a 'depends_on' directive
    invalid values for variant "build_type" in package "ginkgo": ['RelWithDebInfo'] in package 'ginkgo'
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/exago/package.py
```

While it looks like you are right about ginkgo in particular not having that build type, there are additional errors that reference failures in looking for variants `camp`, `magma`, `umpire` and `ginkgo`. These packages are not controlled by their namesake variant, hence my suggestion.